src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT = 10;
+    private static final double STANDART_GROW = 1.5;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT];
+    }
+
     @Override
     public void add(T value) {
-
+        capEnsures();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add element on position " + index);
+        }
+        capEnsures();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
```

Comment:
You already made your array bigger in `capEnsures`, no need to call `arraycopy` twice, just put element in the end

Reply:
I think we should rather remove the `capEnsures()` call from here, just so that we do not have douple array copy, in some cases